cmd/main.go

```diff
@@ -1,128 +1,42 @@
 package main
 
 import (
-	"context"
-	"encoding/json"
-	"io"
-	"io/ioutil"
 	"os"
-	"time"
 
-	"github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/azure"
+	"k8s.io/klog"
+
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/provider"
+
+	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/version"
 
-	log "github.com/sirupsen/logrus"
 	"github.com/spf13/pflag"
 )
 
 var (
-	attributes     = pflag.String("attributes", "", "volume attributes")
-	secrets        = pflag.String("secrets", "", "node publish ref secret")
-	targetPath     = pflag.String("targetPath", "", "Target path to write data.")
-	permission     = pflag.String("permission", "", "File permission")
-	debug          = pflag.Bool("debug", false, "sets log to debug level")
-	versionInfo    = pflag.Bool("version", false, "prints the version information")
-	contextTimeout = pflag.Int("context-timeout", 110, "context timeout in seconds for provider calls")
```

Comment:
How do we treat context timeout here for grpc server and client in the driver?

Reply:
The client in CSI driver passes the kubelet context to the gRPC server as part of the RPC call (https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/pkg/secrets-store/provider_client.go#L104) . The server passes the context to all the underlying functions.

Reply:
do we want to set timeouts for client and server?

Reply:
The parent context is the kubelet context which has a timeout of 2m3s. The same context is propagated across as part of the RPC call. We pass the context to all Azure API calls as well. So the parent context timeout/deadline is essentially the value we want to honor in the provider. The upper ceiling for the timeout is how long kubelet can wait for the driver-provider communication to complete. I don't see a scenario where we would want to override that context with a shorter time. WDYT?

Reply:
okay that makes sense!